File: csharp/main/LiteralUUID.cs
using System;

namespace Deltix.Luminary
{
	// ReSharper disable once InconsistentNaming
	public sealed class LiteralUUID : Literal, ILiteral<Guid>
	{
		public static Guid MinValue = Guid.Empty;
		public static Guid MaxValue = new Guid(-1, -1, -1, Byte.MaxValue, Byte.MaxValue, Byte.MaxValue, Byte.MaxValue, Byte.MaxValue, Byte.MaxValue, Byte.MaxValue, Byte.MaxValue);

		public LiteralUUID(Guid value)
			: base(LiteralKind.UUID)
		{
			Value = value;
		}

		public Guid Value { get; }

		public override String ToString()
		{
			return Value.ToString();
		}
	}
}

File: java/src/main/java/deltix/luminary/LiteralUUID.java
package deltix.luminary;

import java.util.UUID;

public class LiteralUUID extends Literal {
    private final UUID value;
    private final String string;

    public LiteralUUID(UUID value) {
        super(LiteralKind.UUID);
        this.value = value;
        this.string = value.toString().toUpperCase();
    }

    public UUID getValue() {
        return value;
    }

    @Override
    public String toString() {
        return string;
    }

    public static final UUID MIN_VALUE = new UUID(0L, 0L);

    public static final UUID MAX_VALUE = new UUID(-1L, -1L);
}
